src/main/scala/net/snowflake/spark/snowflake/ServerConnection.scala

```diff
@@ -274,6 +290,36 @@ private[snowflake] object ServerConnection {
     jdbcConnectionCount.incrementAndGet()
     conn
   }
+
+  object providedConnections {
+    // use provided JDBC connection
+    private val providedConn = mutable.Map[String, Connection]()
```

Comment:
This is the staging area to cache Snowflake connection. Since Spark API only accepts String values, we can't directly pass JDBC connection to Spark connector.
Instead of that, we cache the connection in the staging area and add the connection id into the parameter map. Therefore, connector can load the connection from the staging area.